### DISCUSSION TITLE
Singlestore JDBC driver support

### DISCUSSION THREAD

**Discussion by msal-priv** [dbeaver/dbeaver#20699] [General]
**Is your feature request related to a problem? Please describe.**
Recently Singlestore released their own JDBC driver - https://github.com/memsql/S2-JDBC-Connector
Singlestore connection is still MariaDB/MySQL compatible and can be used but in some cases it is already preferred to use the official vendor Driver.
Additionally the Driver has some specific to Singlestore parameters that can be used (see details in links below).

**Describe the solution you'd like**
Ability to choose Singlestore in UI such as New Connection wizard UI and in Driver Manager without manually adding it to the Driver Manager.

**Describe alternatives you've considered**
Adding the driver manually with adding the relevant parameters in the URL (requires some understanding JDBC parameters).
Using MariaDB/MySQL drivers as alternative.

**Additional context**
https://github.com/memsql/S2-JDBC-Connector/releases
https://github.com/memsql/S2-JDBC-Connector
https://docs.singlestore.com/managed-service/en/developer-resources/connect-with-application-development-tools/connect-with-java-jdbc/the-singlestore-jdbc-driver.html
https://docs.singlestore.com/managed-service/en/connect-to-your-workspace/connect-with-dbeaver.html

**Adding the Driver manually test**
For NON SSL/SSO connection example (using DBeaver CE v22.3.3 and Singlestore JDBC driver v1.1.4 vs Singlestore v8.0.7): 
![image](https://user-images.githubusercontent.com/77410911/216585476-67792263-73e3-4770-b786-7d1ff27adfa5.png)
![image](https://user-images.githubusercontent.com/77410911/216585650-705c9f2b-fa62-4461-ab7a-2d7a9b3c3c6f.png)

**Comment by LonwoLonwo** [comment]
Starting version 23.1.0, DBeaver supports SingleStore in PRO products.
You can try it for free: https://dbeaver.com/trial/
